src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs

```diff
@@ -152,5 +152,50 @@ public static partial class Activator
         }
 
         private static T CreateDefaultInstance<T>() where T : struct => default;
+
+        public static Func<object?> CreateFactory([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] Type type, bool nonPublic)
+        {
+            if (type is null)
+                throw new ArgumentNullException(nameof(type));
+
+            if (type.UnderlyingSystemType is not RuntimeType rt)
+                throw new ArgumentException(SR.Arg_MustBeType, nameof(type));
+
+            // First create the factory instance.
+
+            ActivationFactory factory = new ActivationFactory(rt);
+            if (!nonPublic && !factory.CtorIsPublic)
+            {
+                throw new MissingMethodException(SR.Format(SR.Arg_NoDefCTor, rt));
+            }
+
+            // Then create a delegate to factory.CreateInstance, closed over the "this" parameter.
+
+            return (Func<object?>)factory.CreateDelegate((RuntimeType)typeof(Func<object?>));
+        }
+
+        [DynamicDependency("#ctor", typeof(ActivationFactory<>))]
+        public static Func<T> CreateFactory<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]T>()
+        {
+            if (typeof(T).IsValueType)
+            {
+                IActivationFactory factory = (IActivationFactory)RuntimeTypeHandle.CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(ActivationFactory<>), (RuntimeType)typeof(T));
```

Comment:
CreateInstanceForAnotherGenericParameter is AOT unfriendly API that depends on JIT. Can this be just something something like this?
```
class StructFactory<T>
{
    T CreateInstance()
    {
         T value;
         _pfnConstructor(&value);
         return value;
    }
}
```
return new StructFactory<T>().;
```
?